src/server/transport/socketio.ts

```diff
@@ -70,9 +72,9 @@ export function TransportAPI(
   /**
    * Send a message to all clients.
    */
-  const sendAll: MasterTransport['sendAll'] = (makePlayerData) => {
```

Comment:
We also need to refactor the `send` method. It is currently only used by the `sync` action, but that is still filtering in the Master: https://github.com/vdfdev/boardgame.io/blob/38a62267ce23508f82596557e3020b78ab4e0c36/src/master/master.ts#L371-L395

(Not sure if the `send` method is such an issue for Pub/Sub implementation, but it’s probably a good idea to refactor both together for consistency nonetheless.)

Reply:
Done, good catch. I didnt notice this before because I was only concerned with the `sendAll` (`send` should not need pub sub as we dont need to propagate to multiple servers), but It looks much cleaner now that `Master` only cares about the `IntermediateTransportData` and the Transport cares about the `TransportData`